### DISCUSSION TITLE
Resolving an array of components in config

### DISCUSSION THREAD

**Discussion by narayanacharya6** [explosion/spaCy#9252] [Help: Coding & Implementations] [answered]
I have a component that takes in an array of other misc registered functions. Something like this

```
class CustomComponent(Pipe):

    def __init__(self, nlp, name, others: Iterable[Callable[[X], X]]):
        self.others = others
```

The `others` above is basically an array of functions registered with the spaCy misc registry.

```
@spacy.registry.misc('func1')
def make_func_1(func_specific_config_params):
    return ...

@spacy.registry.misc('func2')
def make_func_2(func_specific_config_params):
    return ...
```

I try to use these as:

```
nlp = spacy.blank('en')
nlp.add_pipe('CustomComponent'), config={"others": [{"@misc": "func1"}, {"@misc", "func2"}]})
```

Obviously this does not work because the `others` passed via the config resolves to a list of dictionaries. How can I change this implementation to allow the config to resolve to an array of spaCy registered functions?

**Comment by narayanacharya6** [comment, marked as answer]
Please ignore above. I ended up wrapping the array in another component class to resolve the issue.

**Reply by polm** [reply]
Glad you figured it out! Just want to say I think that's the right way to handle this situation.

**Reply by alejandrods** [reply]
Hi, I am facing the same problem but I could not figure out how to solve it. Could you provide an example code, please?

**Reply by narayanacharya6** [reply]
Continuing from the same example from above, you can have another registered function that returns an array of components you need. Something like this:

```
@spacy.registry.misc('func1')
def make_func_1(func_specific_config_params):
    return ...

@spacy.registry.misc('func2')
def make_func_2(func_specific_config_params):
    return ...
   
@spacy.registry.misc('combined')
def make_combined(func_specific_config_params):
    return [make_func_1(...), make_func_2(...)]
```

Then to use the same:
```
nlp = spacy.blank('en')
nlp.add_pipe('CustomComponent'), config={"others": {"@misc": "combined"}})
```

Hope this helps :)
